src/components/VersionedJavaDocLink.tsx

```diff
@@ -0,0 +1,21 @@
+import React, {useEffect, useState} from "react";
+import SoftwareVersionFetcher from "../minecraft-versioning/SoftwareVersionFetcher";
+
+export default function VersionedJavaDocLink({ target, project = "paper", children }) {
+    const [href, setHref] = useState(null);
+
+    useEffect(() => {
+        (async () => {
+            let version = await SoftwareVersionFetcher.getMajorVersion(project);
+            if (project === "velocity") {
+                version = version[0] + ".0.0";
+            }
+
+            setHref("https://jd.papermc.io/" + project + "/" + version + target);
+        })();
+    }, [target]);
+
+    return (
```

Comment:
Does this need to have a return early if href isn't set yet?

Reply:
Not quite sure what you mean here, could you elaborate please?

Reply:
idk, just looking at the pattern here https://github.com/PaperMC/docs/pull/212/files#diff-15117eb1f365fc91bf29f3791c31fab4d45ec0af8b3238dbf474e0bcc800526cR25

Reply:
My thought process here was that you could just return the `<a>` tag regardless and the href would just load in the background. maybe one with an empty href so that you don't end up with a duff link whilst it loads?

Reply:
ya, that's fine then.